Sources/PackageGraph/Pubgrub.swift

```diff
@@ -456,9 +481,9 @@ final class PartialSolution {
 
     /// Create a new decision assignment and add it to the partial solution's
     /// list of known assignments.
-    func decide(_ package: PackageReference, atExactVersion version: Version) {
+    func decide(_ package: PackageReference, at version: BoundVersion) {
```

Comment:
Can you make another PR that only contains the `BoundVersion` API changes? That will help with simplifying the diffs for the branch-based dependencies.